data/provider/src/test/java/tech/pegasys/artemis/api/ChainDataProviderTest.java

```diff
@@ -315,4 +318,78 @@ void getStateByBlockRoot_shouldReturnBeaconStateWhenFound()
     BeaconState result = future.get().get();
     assertThat(result).usingRecursiveComparison().isEqualTo(beaconState);
   }
+
+  @Test
+  void getUnsignedAttestationAtSlot_shouldReturnEmptyIfStoreNotFound() {
+    ChainDataProvider provider =
+        new ChainDataProvider(chainStorageClient, mockCombinedChainDataClient);
+    when(mockCombinedChainDataClient.isStoreAvailable()).thenReturn(false);
+    Optional<Attestation> optional = provider.getUnsignedAttestationAtSlot(ZERO, 0);
+    verify(mockCombinedChainDataClient).isStoreAvailable();
+    assertTrue(optional.isEmpty());
+  }
+
+  @Test
+  void getUnsignedAttestationAtSlot_shouldReturnEmptyIfSlotIsFinalized() {
+    ChainDataProvider provider =
+        new ChainDataProvider(chainStorageClient, mockCombinedChainDataClient);
+    when(mockCombinedChainDataClient.isStoreAvailable()).thenReturn(true);
+    when(mockCombinedChainDataClient.isFinalized(ZERO)).thenReturn(true);
+    assertThrows(
+        IllegalArgumentException.class, () -> provider.getUnsignedAttestationAtSlot(ZERO, 0));
+    verify(mockCombinedChainDataClient).isStoreAvailable();
+    verify(mockCombinedChainDataClient).isFinalized(ZERO);
+  }
+
+  @Test
+  void getUnsignedAttestationAtSlot_shouldReturnEmptyIfCommitteeBelowRange() {
+    ChainDataProvider provider =
+        new ChainDataProvider(chainStorageClient, mockCombinedChainDataClient);
+    when(mockCombinedChainDataClient.isStoreAvailable()).thenReturn(true);
+    when(mockCombinedChainDataClient.isFinalized(ZERO)).thenReturn(false);
+    assertThrows(
+        IllegalArgumentException.class, () -> provider.getUnsignedAttestationAtSlot(ZERO, -1));
+    verify(mockCombinedChainDataClient).isStoreAvailable();
+    verify(mockCombinedChainDataClient).isFinalized(ZERO);
+  }
+
+  @Test
+  void getUnsignedAttestationAtSlot_shouldReturnEmptyIfCommitteeAboveRange() {
+    ChainDataProvider provider =
+        new ChainDataProvider(chainStorageClient, mockCombinedChainDataClient);
+    when(mockCombinedChainDataClient.isStoreAvailable()).thenReturn(true);
+    when(mockCombinedChainDataClient.isFinalized(ZERO)).thenReturn(false);
+    assertThrows(
+        IllegalArgumentException.class, () -> provider.getUnsignedAttestationAtSlot(ZERO, 1));
+    verify(mockCombinedChainDataClient).isStoreAvailable();
+    verify(mockCombinedChainDataClient).isFinalized(ZERO);
```

Comment:
these 3 tests are almost the same - wonder if we can refactor to make the differences clear?

Reply:
refactored.